components/src/utils/mutations.ts

```diff
@@ -1,20 +1,33 @@
 import { type MutationType, type SequenceType } from '../types';
 
 export interface Mutation {
+    readonly position: number;
+    readonly code: string;
+    readonly type: MutationType;
+    readonly segment?: string;
+}
+
+export interface MutationClass {
     readonly segment: string | undefined;
     readonly position: number;
     readonly code: string;
     readonly type: MutationType;
```

Comment:
```suggestion
export interface MutationClass extends Mutation {
```
Doesn't that do the job?